### ISSUE TITLE
[Feature] Ayu themes

### ISSUE THREAD

**Issue by seagle0128** [hlissner/emacs-doom-themes#411]
Ayu color scheme seems very clean and elegant. Can you please integrate it to `doom-themes`?

Reference:
https://github.com/ayu-theme/ayu-colors
https://github.com/nonissue/.emacs.d/blob/master/themes/ayu-theme.el

![ayu](https://camo.githubusercontent.com/7925fe5eb9bdc44e00a3129c7c6f5799bb1dbfb1/687474703a2f2f692e696d6775722e636f6d2f623365744251582e706e67)

**Comment by ianpan870102** [comment]
Second! Ayu is beautiful.

**Comment by LoveSponge** [comment]
Hi, I third this - so i'm trying to make it ; 
I've copy/pasted a file in /themes directory and renamed it, started changing some values etc... how do i test it? i've tried to set the theme in config but the name is not recognised - i think i need to build/compile `doom-themes` but i've got no idea how (i've tried `doom sync` but it prints nothing to compile, so i guess this is wrong usage for this command)...

i'm a n00b with emacs and lisp in general but i'd really like to get this implemented, any help is appreciated.

Cheers,
LS

**Comment by Emiller88** [comment]
@LoveSponge You can do like I did here. https://github.com/Emiller88/doom-emacs-private/blob/master/themes/doom-ayu-light-theme.el

The file naming and location is important. 

I'm struggling with the colors because they're inconsistant in their colors across different editor themes.

**Comment by LoveSponge** [comment]
so i've sort of got something, they require some custom stuff though so for different languages it could be kind of ugly... i'll submit a pr anyway to get the code out there.

**Comment by ema2159** [comment]
Almost complete! The only one missing would be Ayu-dark. @LoveSponge, given that you're familiar with this set of themes, do you feel like implementing the dark variant?

**Comment by LoveSponge** [comment]
@ema2159 honestly I barely had the time to get the other two over the line, I have the scaffolding set out, I think there was just an issue where certain org highlights were using ayu white colours (in mirage theme) for some reason...
I'll see if I can't fix that one up and then use the template to copy /paste for ayu-dark

**Comment by ejez** [comment]
Thanks for the work on this theme!

I would like to report this issue on org mode:
![20210223_12h11m51s](https://user-images.githubusercontent.com/48388358/108822922-d1e49d00-75d0-11eb-8ae2-87e8e06957af.png)

Source code blocks have a white background.

**Comment by seagle0128** [comment]
I tried the themes. One suggestion, the line numbers are too bright (or dark). Thoughts?

**Comment by ianpan870102** [comment]
> I tried the themes. One suggestion, the line numbers are too bright (or dark). Thoughts?

I agree. Perhaps using the face of "comment" for "line-number" can work well?

**Comment by seagle0128** [comment]
FYI

Some screenshots: https://github.com/ayu-theme/vscode-ayu

**Comment by LoveSponge** [comment]
Thanks for all the feedback, @ejez @seagle0128 @ianpan870102 

Unfortunately i've not spent much time since the last commit;

With regards to your actual comments:
Line numbers: Agreed, I can modify them.
vscode styling: Agreed, i was actually using vscode as much as possible to copy the colours.
Org mode code style bug: It's known, but i've got no idea how to fix it, strangely in ayu-light it's not an issue, what is going wrong?

Any help on these would be appreciated, the code itself is really simple, but i know very little about emacs-lisp and moreover only a tip of the iceberg on how emacs themes work. (i used `M-x describe-theme` ALLOT.)

**Comment by mattsawyer77** [comment]
Thanks for the work on this nice theme. I noticed that the ayu-mirage theme is already available(?) I can use it -- though each time I open emacs, it asks me whether I trust the theme.

**Comment by ianpan870102** [comment]
> though each time I open emacs, it asks me whether I trust the theme.

Did you provide an argument of `t`? Like so: `(load-theme 'doom-ayu-mirage t)`

**Comment by mattsawyer77** [comment]
aha, thank you @ianpan870102, I had indeed forgotten the `t`.

**Comment by gnuyent** [comment]
Hi, thanks for the hard work! I looked at some other themes (specifically [Iosvkem](https://github.com/hlissner/emacs-doom-themes/blob/master/themes/doom-Iosvkem-theme.el)) and they use the following for highlighting org-mode code blocks:

```emacs-lisp
 ((org-block &override) :background COLOR_HERE)            ;; highlight the code block
 ((org-quote &override) :background COLOR_HERE)            ;; highlight the quote block
 ((org-block-begin-line &override) :background COLOR_HERE) ;; highlight begin_src and end_src
```

Other themes seem to have these as the same color.

I've been playing around with some ayu-dark colors with `bg-alt` for `COLOR_HERE` and it looks quite nice.

![image](https://user-images.githubusercontent.com/10103146/114347537-eeee1480-9b19-11eb-9f4b-fc410280da4b.png)

**Comment by LoveSponge** [comment]
Thank you all for your awesome feedback, and the help with this, I've just made some ammendments, focusing on org-mode specifically... Whatcha think?

Special thanks to @gnuyent for the org-x-syntax names!

Changes:
- Line numbers
- Org mode faces
- Ayu Dark

MIRAGE

![image](https://user-images.githubusercontent.com/12626802/119645655-f138d380-be26-11eb-9718-bc997e365a3f.png)

LIGHT

![image](https://user-images.githubusercontent.com/12626802/119645546-ce0e2400-be26-11eb-87dd-c90e243b5bd4.png)

 [NEW] DARK

![image](https://user-images.githubusercontent.com/12626802/119646397-c733e100-be27-11eb-89b5-323be8a42dfd.png)

**Comment by LoveSponge** [comment]
@seagle0128 thanks!

I'm actually still making tweaks to a few things, feel free to clone my repository, I don't want to bother the doom devs with a PR until it's actually ready...

Also I came across a bug where some specific magit screen was showing the hunk diff with a white background (similar to how org was doing) but I can't see where... If anyone knows or runs into it please let me know!

Cheers,
LS

**Comment by gf3** [comment]
the lsp highlight colour in ayu light can be hard to read, however i'm pretty new to emacs and i don't know how to fix this:

[GIF](https://screenshot.pics/s/screencast_2021-10-15_18-53-40.gif)

**Comment by seagle0128** [comment]
When the new implementation will be merged?

**Comment by JP-Ellis** [comment]
@LoveSponge Have you made much more progress on Ayu dark?  I am very keen on seeing integrated in Doom :smile:

**Comment by MONO77** [comment]
Hi I am wondering if we can add a contrast mode to the theme? The current theme is a bit hard to distinguish between split panes.

![image](https://user-images.githubusercontent.com/44114934/175183402-25057765-a0e6-48ae-a3f4-567eda032c90.png)

**Comment by seagle0128** [comment]
@MONO77 You may need `window-divider-mode`.

**Comment by MONO77** [comment]
Hi @seagle0128 thanks for your tip. However, I don't think`window-driver-mode` solves the problem, at least with the `ays-light`. Here is a screenshot:
![image](https://user-images.githubusercontent.com/44114934/175355559-041764b0-2458-4f9b-a086-7f420d01f421.png)
If you zoom to an extent, there is a really light split that can be barely seen. 
However, what I was suggesting is making different panes (eg. file explorer and code) having different degree of darkness (like the one in my last post). 

Let me use another example to demonstrate:
![image](https://user-images.githubusercontent.com/44114934/175356714-0cbed9e1-883f-4aab-a398-2186f3422d00.png)
![image](https://user-images.githubusercontent.com/44114934/175356130-b38f6104-a31b-45df-bcba-82f8ace3585b.png)
The two screenshots are using Material Darker theme and Material Darker High Contrast respectively in VSCode. As you can see, the second one obviously have more contrast between different panes, which may not be that aesthetically appealing, but is more usable imo.

**Comment by seagle0128** [comment]
@MONO77 Well, basically I agree with you. `window-divider-mode` is just a workaround before this is fixed. You can customize `window-divider` face.

**Comment by axwnet** [comment]
Window dividers should be roughly #C5C7C9

I think a change on the following line is required: https://github.com/doomemacs/themes/blob/cc7686783e5e6e4174a345c768047d524f0dd54f/themes/doom-ayu-light-theme.el#L76

common bg is #fafafa

setting it to common-fg probably gives better contrast

will have to try first before sending a fix

**Comment by seagle0128** [comment]
The colors of line numbers in `doom-ayu-dark` and `doom-ayu-mirage` are too light.
